### PR TITLE
Improve Validation

### PR DESCRIPTION
**Pull request by pfefferle**
This PR is about a leaner validation method so that not every handler class has to be validated again and again.

## Proposed changes:

* Disable `ACTIVITYPUB_DISABLE_INCOMING_INTERACTIONS` on Inbox level
* Improve `args` validation, to validate the Activity-Object as early as possible
* ...

### Other information:

- [ ] Have you written new tests for your changes, if applicable?

## Testing instructions:

* Go to '..'
*

### CODE REVIEW

**Review by mattwiebe — APPROVED** [review]
I don't understand enough about validation challenges, but this approach and your reasoning seems sound.

File: includes/handler/class-create.php [hunk 1]
```diff
@@ -21,6 +21,13 @@ public static function init() {
 			10,
 			3
 		);
+
+		\add_filter(
```
Comment (Menrath) [inline includes/handler/class-create.php:25]: @pfefferle Am I getting something wrong or is this filter added every time? That would also mean, that if one adds a custom handler for a `Create` activity one must also add a custom validation function that completely overrides (is applied later, higher priority) the one in this class (if for example one does not need `inReplyTo` set).

Comment (pfefferle) [reply includes/handler/class-create.php:25]: good point!!!

Comment (Menrath) [reply includes/handler/class-create.php:25]: @akirk Does this not prevent your friends plugin from receiving any creates to your create handler?

Comment (akirk) [reply includes/handler/class-create.php:25]: Maybe I am reading this wrong but I am listening on `activitypub_inbox_create` which should still be called normally, no?

https://github.com/akirk/friends/blob/199af50ae54ee82e8cc77b2311a7149fb1eea8b2/feed-parsers/class-feed-parser-activitypub.php#L45

Comment (Menrath) [reply includes/handler/class-create.php:25]: @akirk This is what my original question was about. I am not totally sure. I have been studying these lines of your code recently, before starting implementing my event-import feature where [I also added an action for `activitypub_inbox_create`](https://code.event-federation.eu/Event-Federation/wordpress-event-bridge-for-activitypub/src/commit/16762b2b3140759e963faab10db36e47b8023f7c/includes/activitypub/handler/class-create.php#L22). But within my setup the "aggressive" filter here makes incoming `Create` activities not even reach the handler in `activitypub_inbox_create` when for example the `object` within the `Create` does not have `inReplyTo` set. So I was wondering, if I am missing something else, or something is special in my setup etc.

**Pull request merged**